### PR TITLE
[ONNX] Add test_fx_op_consistency.py

### PR DESCRIPTION
**Pull request by titaiwangms**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #100173
* #100172
* __->__ #99465

Add op consistency test for fx exporter. There will be another PR to work around the limitations of https://github.com/pytorch/pytorch/issues/99534

### CODE REVIEW

File: test/onnx/test_op_consistency.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def xfail(
     )
 
 
-def dont_care(
```
Comment (titaiwangms) [inline test/onnx/test_op_consistency.py:147]: dont_care is inconsistent with what we have in onnx-script

File: test/onnx/test_fx_op_consistency.py [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+                inputs=repr(inputs),
+                kwargs=repr(cpu_sample.kwargs),
+            ):
+                skip_reason = _should_skip_test_sample(op.name, cpu_sample)
+                if skip_reason is not None:
+                    # Cannot use self.skip because pytest would skip the entire test
+                    warnings.warn(f"skipped sample {i}. Reason: {skip_reason}")
+                    continue
+
+                model = SingleOpModel(op.op, cpu_sample.kwargs)
```
Comment (titaiwangms) [inline test/onnx/test_fx_op_consistency.py:432]: dynamo currently doesn't support CustomObj, so I have to pour op out.

File: test/onnx/test_fx_op_consistency.py [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+#        e.g. the test is flaky or some tests pass. Otherwise, use xfail.
+#    Use skip if we don't care about the test passing, e.g. ONNX doesn't support the usage.
+#    Use xfail if a test fails now and we want to eventually fix the test.
+EXPECTED_SKIPS_OR_FAILS: Tuple[DecorateMeta, ...] = (
+    skip(
+        "ceil", dtypes=BOOL_TYPES + INT_TYPES,
+        reason=reason_onnx_does_not_support("Ceil")
+    ),
+    fixme("ceil", dtypes=[torch.float64], reason=reason_onnx_runtime_does_not_support("Ceil", ["f64"])),
+    fixme("unflatten", reason="AssertionError: Expected 1 inputs, got 3"),
```
Comment (titaiwangms) [inline test/onnx/test_fx_op_consistency.py:328]: Constantly seeing this error also proves this test could be useful.

Comment (titaiwangms) [reply test/onnx/test_fx_op_consistency.py:328]: should be xfail

Comment (titaiwangms) [reply test/onnx/test_fx_op_consistency.py:328]: https://github.com/pytorch/pytorch/issues/99534

File: test/onnx/test_fx_op_consistency.py (hunk 2 above)
Comment (titaiwangms) [reply test/onnx/test_fx_op_consistency.py:432]: @justinchuby

**Review by justinchuby — APPROVED** [review]

File: test/onnx/test_fx_op_consistency.py [hunk 4]
```diff
@@ -32,13 +32,16 @@
 import parameterized
 
 import torch
+from test_fx_to_onnx_with_onnxruntime import (
+    _run_test_with_fx_to_onnx_exporter_and_onnx_runtime,
```
Comment (justinchuby) [inline test/onnx/test_fx_op_consistency.py:36]: Avoid importing from a test file. Prefer to extract this function out to a common file and mark as non private?

Comment (justinchuby) [reply test/onnx/test_fx_op_consistency.py:36]: We should avoid importing private functions.

Comment (titaiwangms) [reply test/onnx/test_fx_op_consistency.py:36]: I willl update it in https://github.com/pytorch/pytorch/pull/99434

**Pull request closed without merging**